internal/ingest/ingest.go

```diff
@@ -71,7 +73,18 @@ func NewLegIngester(ctx context.Context, cfg config.Ingest, h host.Host,
 	idxr *indexer.Engine, reg *registry.Registry, ds datastore.Batching) (LegIngester, error) {
 
 	lsys := mkLinkSystem(ds, reg)
-	lms, err := legs.NewMultiSubscriber(ctx, h, ds, lsys, cfg.PubSubTopic)
+
+	// Construct a selector that recursively looks for nodes with field
+	// "PreviousID" as per Advertisement schema.
+	ssb := builder.NewSelectorSpecBuilder(basicnode.Prototype.Any)
+	advAndChunkSel := ssb.ExploreFields(
+		func(efsb builder.ExploreFieldsSpecBuilder) {
+			efsb.Insert("Entries", ssb.ExploreRecursiveEdge())
```

Comment:
i thought we didn't want the basic advertisement sync/selector to include the fetch of entries / next.

Reply:
The rationale is to have that work done in a separate PR because it's beyond a simple dependency upgrade and involves changing the way storage hooks work among other changes.

I have explained more here: https://github.com/filecoin-project/storetheindex/pull/105#issuecomment-979249687